statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vcov : square ndarray
+            The variance-covariance matrix of the estimates, obtained using
+            the (drop 1) jackknife procedure.
+        pseudo : ndarray
+            The jackknife pseudo-values.
+        """
+        est = self._stat(self.design.weights)
+
+        jdata = []
+        try:
+            k = self.design.nclust
+        except AttributeError:
+            k = self.design.rep_weights.shape[1]
+        # for each cluster
+        for c in range(k):
+            # get jackknife weights
+            w = self.design.get_rep_weights(c=c)
+            jdata.append(self._stat(w))
+        jdata = np.asarray(jdata)
+        if self.mse:
+            print('mse specified')
+            jdata -= est
+        else:
+            if self.design.rep_weights is None:
+                for s in range(self.design.nstrat):
+                    # center the 'delete 1' statistic
+                    jdata[self.design.ii[s], :] -= jdata[self.design.ii[s],
+                                                         :].mean(0)
+            else:
+                jdata -= jdata.mean(0)
+
+        if self.design.rep_weights is None:
+            print("extra factors")
+            nh = self.design.ncs[self.design.sfclust].astype(np.float64)
+            _pseudo = jdata + nh[:, None] * (np.dot(self.design.weights,
+                                                self.data) - jdata)
+            mh = np.sqrt((nh - 1) / nh)
+            fh = np.sqrt(1 - self.design.fpc)
+            jdata = fh[:, None] * mh[:, None] * jdata
+        else:
+            nh = self.design.rep_weights.shape[1]
+            print(nh)
+            mh = np.sqrt((nh - 1) / nh)
+            jdata *= mh
+        vcov = np.dot(jdata.T, jdata)
+
+        return est, np.sqrt(np.diag(vcov))
+
+
+class SurveyMean(SurveyStat):
+    """
+    Calculates the mean for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, mse=False, replicates=None, bsn=None):
+        super().__init__(design, mse)
+        self.data = np.asarray(data)
+        if self.design.se_method == "jack":
+            self.est, self.std = self._jackknife()
+        else:
+            self.est, self.std = self._bootstrap(replicates, bsn)
+
```

Comment:
Maybe better to add this upstream in SurveyStat, so the subclasses don't each have to do it.